tests/cases/compiler/dependentReturnType7.ts

```diff
@@ -0,0 +1,29 @@
+// @strict: true
+// @noEmit: true
+// @target: esnext
+// @checkJs: true
+// @filename: file.js
+
+/** @type {Map<string, string>} */
+const sources = new Map();
+/**
+
+ * @param {string=} type the type of source that should be generated
+ * @returns {String}
+ */
+function source(type = "javascript") {

```

Comment:
what is this testing?

Reply:
This is also from an earlier version of the code that had a bug that only showed in extended tests. The bug was looking inside a conditional return expression that has a JSDoc type cast, which we shouldn't do. But I should move this to a different test and add comments.